src/Plugins/RestServer/Controllers/v1/ContractsController.cs

```diff
@@ -0,0 +1,220 @@
+// Copyright (C) 2015-2024 The Neo Project.
+//
+// ContractsController.cs file belongs to the neo project and is free
+// software distributed under the MIT software license, see the
+// accompanying file LICENSE in the main directory of the
+// repository or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Neo.Plugins.RestServer.Exceptions;
+using Neo.Plugins.RestServer.Extensions;
+using Neo.Plugins.RestServer.Helpers;
+using Neo.Plugins.RestServer.Models;
+using Neo.Plugins.RestServer.Models.Error;
+using Neo.SmartContract;
+using Neo.SmartContract.Manifest;
+using Neo.SmartContract.Native;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Mime;
+
+namespace Neo.Plugins.RestServer.Controllers.v1
+{
+    [Route("/api/v{version:apiVersion}/contracts")]
+    [Produces(MediaTypeNames.Application.Json)]
+    [Consumes(MediaTypeNames.Application.Json)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest, Type = typeof(ErrorModel))]
+    [ApiVersion("1.0")]
+    [ApiController]
+    public class ContractsController : ControllerBase
+    {
+        private readonly NeoSystem _neosystem;
+
+        public ContractsController()
+        {
+            _neosystem = RestServerPlugin.NeoSystem ?? throw new NodeNetworkException();
+        }
+
+        /// <summary>
+        /// Get all the smart contracts from the blockchain.
+        /// </summary>
+        /// <param name="skip" example="1">Page</param>
+        /// <param name="take" example="50">Page Size</param>
+        /// <returns>An array of Contract object.</returns>
+        /// <response code="204">No more pages.</response>
+        /// <response code="200">Successful</response>
+        /// <response code="400">An error occurred. See Response for details.</response>
+        [HttpGet(Name = "GetContracts")]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(ContractState[]))]
+        public IActionResult Get(
+            [FromQuery(Name = "page")]
+            int skip = 1,
+            [FromQuery(Name = "size")]
+            int take = 50)
+        {
+            if (skip < 1 || take < 1 || take > RestServerSettings.Current.MaxPageSize)
+                throw new InvalidParameterRangeException();
+            var contracts = NativeContract.ContractManagement.ListContracts(_neosystem.StoreView);
+            if (contracts.Any() == false)
+                return NoContent();
+            var contractRequestList = contracts.OrderBy(o => o.Manifest.Name).Skip((skip - 1) * take).Take(take);
```

Comment:
```suggestion
            var contractRequestList = contracts.OrderBy(o => o.Id).Take(take);
```
Seems it's more making sense by ordering by id. It's much easier to get all native contracts' manifest.

Reply:
Will need the `skip` for paging

Reply:
> Will need the `skip` for paging

Yes. But if make it as ` contracts.OrderBy(o => o.Id).Skip((skip - 1).Take(take);`, when page is 1, it will skip contractid `0`. That's a problem. Need a better way.